pkg/ddc/thin/engine.go

```diff
@@ -117,10 +117,21 @@ func Precheck(client client.Client, key types.NamespacedName) (found bool, err e
 func CheckReferenceDatasetRuntime(client client.Client, runtime *datav1alpha1.ThinRuntime) (bool, error) {
 	dataset, err := utils.GetDataset(client, runtime.Name, runtime.Namespace)
 	if err != nil {
-		return false, err
+		if utils.IgnoreNotFound(err) == nil && runtime.Status.Mounts != nil && len(runtime.Status.Mounts) != 0 {
```

Comment:
I think we should make it work even the virtualDataset is deleted.

Reply:
Normally, the virtualDataset would not be deleted because its reference runtime has not been cleaned up.

Reply:
But it may happen when deleting virtualDataset forcely.  How to handle this then?

Reply:
fixed